content/blog/unified-haskell-installer.md

```diff
@@ -0,0 +1,87 @@
+2. Some people really don't like typing in `stack ghc` instead of `ghc`. I'm overall a fan of needing the wrapper tooling, since it allows you to trivially switch which toolchain you're using, and makes it more blatant that certain environment variables (like `STACK_YAML`) and files (like `stack.yaml`) may affect behavior.
+3. Stack isn't the only tool in town. For example, and many people want to use `cabal-install` instead. Stack doesn't have support for downloading `cabal-install` binaries right now. Even if it did, it's not clear that the `cabal-install` team will consent to an installer that leverages Stack like this. But there are even _more_ binaries we may wish to install and manage going forward, such as linters, stylers, IDE integration tools, etc.
+
+But if you're using Stack to build your projects, "how do I install GHC and associated tools" is basically a solved problem. Perhaps (2) [irritates some people](https://github.com/commercialhaskell/stack/issues/5515), but we'll get back to that. Perhaps you don't like `curl`-able scripts, or Windows installers, or user-specific installs. You can go ahead and manually download the Stack executable and use things like `{ system-ghc: true, install-ghc: false }`. But from my experience, most people just use the defaults and are happy enough.
+
+## Everything else
+
+But as mentioned, not everyone is on Stack. And there's other tooling we'd like to be able to install. And we don't want to bifurcate our install methods too much. And it's kind of weird that Stack is both a toolchain manager _and_ a build tool. And frankly, it would be nice to finally get a downloads page that everyone can agree with.
+
+There _are_ other approaches to installing toolchains available today. `ghcup`, Haskell Platform (to some extent), manually `./configure && make install`ing GHC bindists, distro specific methods, etc. Some of these solves some of the three problems I mentioned with Stack. Some solve other problems as well. Some create new problems. But a recurring theme in particular that the Haskell Foundation members have heard is that the story for non-Stack users on Windows is particularly painful.
+
+Let me clarify that I'm wearing two hats with what I'm about to say:
+
+1. A maintainer of Stack and the original author of much of Stack's toolchain install logic, who would always rather see the Stack codebase size go down instead of up.
+2. A member of the Haskell Foundation board, who is looking to improve the Haskell adoption story.
+
+So the problem statement: can't we standardize an install method that meets most people's goal, most of the time?
+
+## The goal
+
+I'll start by laying out the ultimate goal for in my head. This idea's been bouncing around for a few months now, but only came up in May 4, 2021 tech track meeting for the Haskell Foundation. I don't think it's that ambitious of a goal. But in the next section, I'm also going to lay out some shorter term milestones.
```

Comment:
'goal for in my head' is a little awkward.

Reply:
Worse than awkward, it's just wrong. Good catch, thanks!